addons/stock_account/models/stock_quant.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class StockQuant(models.Model):
+    _inherit = 'stock.quant'
+
+    value = fields.Float('Value', compute='_compute_value')
+    currency_id = fields.Many2one(related='product_id.currency_id')
+
+    @api.depends('quantity')
+    def _compute_value(self):
+        """ For standard and AVCO valuation, compute the current accounting
+        valuation of the quants by multiplying the quantity by
+        the standard price. Instead for FIFO, use the quantity times the
+        average cost (valuation layers are not manage by location so the
+        average cost is the same for all location and the valuation field is
+        a estimation more than a real value).
+        """
+        for quant in self:
+            if quant.product_id.cost_method == 'fifo':
+                quantity = quant.product_id.quantity_svl
+                if not quantity:
+                    quant.value = 0.0
+                    continue
+                average_cost = quant.product_id.value_svl / quantity
+                quant.value = quant.quantity * average_cost
+            else:
+                quant.value = quant.quantity * quant.product_id.standard_price
```

Comment:
maybe we could try reading value_svl / quantity_svl out of the loop
``` python
        fifo_products = self.env['product.product']
        standard_avco_products = self.env['product.product']
        for quant in self:
            if quant.product_id.cost_method == 'fifo':
                fifo_products |= quant.product_id
            else:
                standard_avco_products |= quant.product_id
        quantity_svl_per_product = {product: product.quantity_svl for product in fifo_products}
        value_svl_per_product = {product: product.value_svl for product in fifo_products}
        standard_price_per_product = {product: product.standard_price for product in standard_avco_products}

        for quant in self:
            if quant.product_id in fifo_products:
                quantity_svl = quantity_svl_per_product[quant.product_id]
                if float_is_zero(quantity_svl, precision_rounding=quant.product_id.rounding):
                    quant.value = 0
                    continue
                value_svl = value_svl_per_product[quant.product_id]
                quant.value = (value_svl / quantity_svl) * quant.quantity
            else:
                quant.value = quant.quantity * standard_price_per_product[quant.product_id]
```

it doesn't make a difference with the 18quants of the demo data though :(

Reply:
1700ms for 7000 quants(1500 as FIFO) with this code 1400ms for same date without